File: src/main/java/SparkDataType.java
public enum SparkDataType {
    INTEGER,
    FLOAT,
    STRING,
    BOOLEAN;

    public static SparkDataType from(String str) {
        switch (str){
            case "INTEGER": return INTEGER;
            case "FLOAT": return FLOAT;
            case "STRING": return STRING;
            case "BOOLEAN": return BOOLEAN;
            default:
                throw new IllegalArgumentException("Invalid DataType: " + str);
        }
    }
}

File: src/main/scala/SparkDataType.scala
object SparkDataType extends Enumeration {
  type SparkDataType = Value
  val INTEGER, FLOAT, STRING, BOOLEAN = Value

  def from(str: String): SparkDataType.Value = str match {
    case "INTEGER" =>
      INTEGER
    case "FLOAT" =>
      FLOAT
    case "STRING" =>
      STRING
    case "BOOLEAN" =>
      BOOLEAN
    case _ =>
      throw new IllegalArgumentException("Invalid DataType: " + str)
  }
}